addons/stock_account/stock.py

```diff
@@ -323,6 +323,9 @@ def _invoice_create_line(self, cr, uid, moves, journal_id, inv_type='out_invoice
             if is_extra_move[move.id] and invoice_line_vals['product_id'] in product_price_unit:
                 invoice_line_vals['price_unit'] = product_price_unit[invoice_line_vals['product_id']]
             if is_extra_move[move.id]:
+                desc = (inv_type == 'out_invoice' and move.product_id.product_tmpl_id.description_sale) or \
+                    (inv_type == 'in_invoice' and move.product_id.product_tmpl_id.description_purchase)
+                invoice_line_vals['name'] += ' ' + desc if desc else ''
```

Comment:
Invoices of type refund should be considered. I think this hunk should be something like this:

        disc_map = {'out_invoice': move.product_id.description_sale, 
	                'in_invoice': move.product_id.description_purchase, 
	                'out_refund': move.product_id.description_sale, 
	                'in_refund': move.product_id.description_purchase,
                    }
        disc = disc_map[inv_type] 
        invoice_line_vals['name'] += ' ' disc if disc else ''